src/Features/Core/Portable/UseThrowExpression/AbstractUseThrowExpressionDiagnosticAnalyzer.cs

```diff
@@ -143,29 +163,13 @@ private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol
                 throwOperation.Exception.Syntax.GetLocation(),
                 assignmentExpression.Value.Syntax.GetLocation());
 
+            // Report a configurable diagnostic on the throw statement span.
             context.ReportDiagnostic(
-                DiagnosticHelper.Create(Descriptor, throwStatementSyntax.GetLocation(), option.Notification.Severity, additionalLocations: allLocations, properties: null));
-
-            // Fade out the rest of the if that surrounds the 'throw' exception.
+                DiagnosticHelper.Create(s_descriptor, throwStatementSyntax.GetLocation(), option.Notification.Severity, additionalLocations: allLocations, properties: null));
 
-            var tokenBeforeThrow = throwStatementSyntax.GetFirstToken().GetPreviousToken();
-            var tokenAfterThrow = throwStatementSyntax.GetLastToken().GetNextToken();
+            // Report a non-configurable hidden fixable diagnostic on the "if" span so the code fix can be invoked from anywhere inside the "if" span.
```

Comment:
💭 I would also be fine with this part just going away

Reply:
That broke existing unit tests and might also break anyone who is used to triggering the fix from if statement header instead of throw statement. I have tried to be conservative in the sense that only user observable change after this PR would be no fading. I would have probably implemented it as per your suggestion if this was a new analyzer. If required, we can certainly come back and remove the hidden fixable diagnostic in future.

Reply:
i would also prefer this going away.  we really don't generally have the concept of "this is available where there isn't some UI affordance" (which is also why it looks like you had to fight against this class above).  I would just change the tests to say that this does *not* show up there anymore.

THere will be a `...`.  The user will have a message in the task-list.  That's sufficient IMO.

Reply:
Sounds good - given we have more than one vote on just removing this and changing unit tests, I will do the same.

Reply:
thanks!

Reply:
Reverted my original commit and pushed a new commit to remove the fading diagnostic and update unit tests.